eureka-client/src/main/java/com/netflix/discovery/DiscoveryClient.java

```diff
@@ -326,10 +325,10 @@ public synchronized BackupRegistry get() {
             }
 
             DiscoveryJerseyProvider discoveryJerseyProvider = new DiscoveryJerseyProvider(
-                    CodecWrapper.get(CodecType.from(clientConfig.getJsonCodecName())),
-                    CodecWrapper.get(CodecType.from(clientConfig.getXmlCodecName()))
+                    CodecWrappers.getEncoder(clientConfig.getEncoderName()),
```

Comment:
After stabilization period we shall remove all legacy codecs which will leave us with options:
* XML vs JSON (== MediaType)
* full vs compact1 vs compact2 (== scope)
I think public API shall be aligned with this target, while we will have to provide temporary means to specify an exact codec to use.